### PR TITLE
Doc generation using wiremocks

### LINKED ISSUE #165: RESTDocs: Create stubs using  spring-cloud-contract-wiremock
We need to address @dsyer's suggestion on creating stubs using `spring-cloud-contract-wiremock` in RESTDocs

https://github.com/spring-cloud/spring-cloud-skipper/pull/160#issuecomment-335383596

### LINKED ISSUE #672: Refactor documentation tests to leverage Mocks

### PR DESCRIPTION
**Pull request by ilayaperumalg**
- Apply changes from https://github.com/spring-cloud/spring-cloud-skipper/pull/242 from @eddumelendez. Credits to him for the initial PR.
  - Configure the BaseDocumentation to use the Skipper server dependencies conditionally
  - Use mocks to verify the documentation tests
  - Use Mock repositories and JPA repositories based on the nature of the tests
      - If the documentation tests the native repository methods then use the JPA repositories. Otherwise, use the mocks
  - Refactor the existing documentation tests
  - Run the Repository based and other tests in different test suites

Resolves #165
Resolves #672

### CODE REVIEW

File: spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/DefaultReleaseManager.java [hunk 1]
```diff
@@ -226,8 +226,12 @@ private void updateCountProperty(Map<String, Object> model, String appsCount) {
 	}
 
 	public Release status(Release release) {
+		if (release.getInfo().getStatus().equals(StatusCode.DELETED)) {
```
Comment (jvalkeal) [inline spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/DefaultReleaseManager.java]: Btw, why all these changes are needed outside of a tests?

Comment (ilayaperumalg) [reply spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/DefaultReleaseManager.java]: This was part of the changes I had for a different PR that got merged: https://github.com/spring-cloud/spring-cloud-skipper/pull/671

Comment (ilayaperumalg) [reply spring-cloud-skipper-server-core/src/main/java/org/springframework/cloud/skipper/server/deployer/DefaultReleaseManager.java]: @jvalkeal Not at all! Rebased the PR. Thank you!

**Comment by jvalkeal** [comment]
ok going to merge this

**Comment by jvalkeal** [comment]
merged per 2523c96a64cb9298102ce2a05bfd787d21832d11

**Pull request closed without merging**